### PR TITLE
Fix smt solver parameters not getting sent to foundry_show

### PR DESCRIPTION
**Pull request by nwatson22**
Adds `smt-timeout` and `smt-retry-limit` parameters to foundry_show, as they are used in this function to call the `get-model` endpoint. Also attempt to fix `test_foundry_merge_nodes` flakiness by upping the smt parameters to be in line with the rest of the `foundry_prove` tests and adding these parameters to the `foundry_step_node` calls.

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/utils.py [hunk 1]
```diff
@@ -326,6 +326,7 @@ def legacy_explore(
     trace_rewrites: bool = False,
 ) -> Iterator[KCFGExplore]:
     # Old way of handling KCFGExplore, to be removed
+
```
Comment (nwatson22) [inline kevm-pyk/src/kevm_pyk/utils.py:329]:
```suggestion
```

File: kevm-pyk/src/tests/integration/test_foundry_prove.py [hunk 2]
```diff
@@ -105,8 +105,8 @@ def test_foundry_prove(test_id: str, foundry_root: Path, update_expected_output:
         foundry_root,
         tests=[test_id],
         simplify_init=False,
-        smt_timeout=300,
-        smt_retry_limit=10,
+        smt_timeout=50,
+        smt_retry_limit=1,
```
Comment (ehildenb) [inline kevm-pyk/src/tests/integration/test_foundry_prove.py:109]: Why are these reduced?

Comment (nwatson22) [inline kevm-pyk/src/tests/integration/test_foundry_prove.py:109]:
```suggestion
        smt_timeout=300,
        smt_retry_limit=10,
```

Comment (nwatson22) [reply kevm-pyk/src/tests/integration/test_foundry_prove.py]: Didn't mean to commit this change. Reverted.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**